Composer/packages/client/src/recoilModel/dispatchers/navigation.ts

```diff
@@ -46,7 +59,21 @@ export const navigationDispatcher = () => {
     ({ snapshot }: CallbackInterface) => async (dialogId: string, breadcrumb: BreadcrumbItem[] = []) => {
       const projectId = await snapshot.getPromise(projectIdState);
       const designPageLocation = await snapshot.getPromise(designPageLocationState);
-      const currentUri = `/bot/${projectId}/dialogs/${dialogId}`;
+      const dialogs = await snapshot.getPromise(dialogsState);
+      const currentDialog = dialogs.find(({ id }) => id === dialogId);
+
+      let path;
+      if (dialogId !== designPageLocation.dialogId) {
+        // Redirect to Microsoft.OnBeginDialog trigger if it exists on the dialog
+        const beginDialogIndex = currentDialog?.triggers.findIndex(({ type }) => type === SDKKinds.OnBeginDialog);
+        if (typeof beginDialogIndex !== 'undefined') {
```

Comment:
Is it ok if `beginDialogIndex` is `-1`?

Reply:
Good catch. It should be.
```suggestion
        if (typeof beginDialogIndex !== 'undefined' && beginDialogIndex >= 0) {
```

Reply:
I think just checking `>= 0` is enough.